src/index.js

```diff
@@ -0,0 +1,137 @@
+// Temperature & Landscape UI Module
+let isFahrenheit = true;
+
+const convertToCelsius = (f) => Math.round((f - 32) * 5 / 9);
+
+const WeatherUI = {
+  currentTemp: 70, // always stored in Fahrenheit
+
+  updateTemperatureUI() {
+    const tempDisplay = document.getElementById("tempValue");
+    const landscapeImage = document.getElementById("landscapeImage");
+    const caption = document.getElementById("landscapeCaption");
+
+    // Determine what to show
+    const displayTemp = isFahrenheit ? this.currentTemp : convertToCelsius(this.currentTemp);
+    tempDisplay.textContent = `${displayTemp}°${isFahrenheit ? "F" : "C"}`;
+    tempDisplay.className = "";
+
+    // Choose range thresholds based on current unit
+    let tempForRange = isFahrenheit ? this.currentTemp : convertToCelsius(this.currentTemp);
+
+    if ((isFahrenheit && tempForRange >= 80) || (!isFahrenheit && tempForRange >= 27)) {
+      tempDisplay.classList.add("red");
+      landscapeImage.src = "assets/hot.png";
+      caption.textContent = "City landscape - very hot day.";
+    } else if ((isFahrenheit && tempForRange >= 70) || (!isFahrenheit && tempForRange >= 21)) {
+      tempDisplay.classList.add("orange");
+      landscapeImage.src = "assets/warmer.png";
+      caption.textContent = "City landscape - sunny summer day.";
+    } else if ((isFahrenheit && tempForRange >= 60) || (!isFahrenheit && tempForRange >= 15)) {
+      tempDisplay.classList.add("yellow");
+      landscapeImage.src = "assets/warm.png";
+      caption.textContent = "City landscape - blooming spring.";
+    } else if ((isFahrenheit && tempForRange >= 30) || (!isFahrenheit && tempForRange >= 0)) {
+      tempDisplay.classList.add("teal");
+      landscapeImage.src = "assets/cool.png";
+      caption.textContent = "City landscape - rainy cool day.";
+    } else {
+      tempDisplay.classList.add("blue");
+      landscapeImage.src = "assets/cold.png";
+      caption.textContent = "City landscape - cold snowy day.";
+    }
+  },
+
+  increaseTemp() {
+    this.currentTemp += 1;
+    this.updateTemperatureUI();
+  },
+
+  decreaseTemp() {
+    this.currentTemp -= 1;
+    this.updateTemperatureUI();
+  }
+};
+
+// Update the city name in the header live when user types
+document.addEventListener("DOMContentLoaded", () => {
+    const cityNameInput = document.getElementById("cityNameInput");
+    const headerCityName = document.getElementById("headerCityName");
+
+    cityNameInput.addEventListener("input", () => {
+    headerCityName.textContent = cityNameInput.value;
+    });
+
+  // Add temperature control event listeners
+    document.getElementById("increaseTempControl").addEventListener("click", () => {
+        WeatherUI.increaseTemp();
+    });
+
+    document.getElementById("decreaseTempControl").addEventListener("click", () => {
+        WeatherUI.decreaseTemp();
+    });
```

Comment:
Since increasing and decreasing the temperature are very similar, another approach we could take is to combine them into 1 function:
```javascript
const setTemp = (changeInTemp) => {
  this.currentTemp += changeInTemp;
  this.updateTemperatureUI();
};
```
Then where we register the handlers, we could call our new function with a parameter by doing so inside an anonymous function:
```javascript
  const tempIncreaseButton = document.getElementById('increaseTempControl');
  tempIncreaseButton.addEventListener('click', () => { setTemp(1) });

  const tempDecreaseButton = document.getElementById('decreaseTempControl');
  tempDecreaseButton.addEventListener('click', () => { setTemp(-1) });
```